# find the mean for a group (using LaTex)

In [31]:
#import the libraries
import re
import pandas as pd
import latex

Now let's remake the dataframe page 87 of the book **'Statistical Methods and Data Analysis'** (Ott&Longnecker)

In [32]:
df = pd.DataFrame({'Class': [1,2,3,4,5,6,7,8,9,10,11], 
                   'Class Interval': ['16.25-18.75', '18.75-21.25', '21.25-23.75', 
                                      '23.75-26.25', '26.25-28.75', '28.75-31.25',
                                      '31.25-33.75', '33.75-36.25', '36.25-38.75',
                                      '38.75-41.25', '41.25-43.75'],
                  '$f_i$' : [2,7,7,14,17,24,11,11,3,3,1],
                  '$y_i$': [17.5,20.0,22.5,25.0,27.5,30.0,32.5,35.0,37.5,40.0,42.5],
                  '$f_i y_i$' : [35.0,140.0,157.5,350.0,467.5,720.0,357.5,385.0,112.5,120.0,42.5],
                  '$f_i(y_i-$ $𝑦¯)^2$' : [258.781,551.359,284.484,210.219,32.141,30.375,144.547,
                                            412.672,223.172,371.297,185.641]})
df

Class Class Interval  $f_i$  $y_i$  $f_i y_i$  $f_i(y_i-$ $𝑦¯)^2$
0       1    16.25-18.75      2   17.5       35.0             258.781
1       2    18.75-21.25      7   20.0      140.0             551.359
2       3    21.25-23.75      7   22.5      157.5             284.484
3       4    23.75-26.25     14   25.0      350.0             210.219
4       5    26.25-28.75     17   27.5      467.5              32.141
5       6    28.75-31.25     24   30.0      720.0              30.375
6       7    31.25-33.75     11   32.5      357.5             144.547
7       8    33.75-36.25     11   35.0      385.0             412.672
8       9    36.25-38.75      3   37.5      112.5             223.172
9      10    38.75-41.25      3   40.0      120.0             371.297
10     11    41.25-43.75      1   42.5       42.5             185.641

## Assignment:
Calculate the mean for grouped data of the dataframe 'df'

The formula for this --> mean = $\bar y$  $\sim$ $\frac{\sum_i f_i y_i} n$

where: 
- $y_i$ = midpoint of the $i$th class interval
- $n$ = total number of measurements
- $f_i$ = frequency associated with the $i$th class interval
- $\sim$ denotes 'is approximately equal to'

## Solution 

In [33]:
#first let's add a total row to the dataframe: 
df.loc['Total']= df.sum()
df

Class                                     Class Interval  $f_i$  $y_i$  \
0          1                                        16.25-18.75      2   17.5   
1          2                                        18.75-21.25      7   20.0   
2          3                                        21.25-23.75      7   22.5   
3          4                                        23.75-26.25     14   25.0   
4          5                                        26.25-28.75     17   27.5   
5          6                                        28.75-31.25     24   30.0   
6          7                                        31.25-33.75     11   32.5   
7          8                                        33.75-36.25     11   35.0   
8          9                                        36.25-38.75      3   37.5   
9         10                                        38.75-41.25      3   40.0   
10        11                                        41.25-43.75      1   42.5   
Total     66  16.25-18.7518.75-21.2521.25-23.7523.75-26.2526...    100  330.0   

       $f_i y_i$  $f_i(y_i-$ $𝑦¯)^2$  
0           35.0             258.781  
1          140.0             551.359  
2          157.5             284.484  
3          350.0             210.219  
4          467.5              32.141  
5          720.0              30.375  
6          357.5             144.547  
7          385.0             412.672  
8          112.5             223.172  
9          120.0             371.297  
10          42.5             185.641  
Total     2887.5            2704.688

With this information, we can fill in the formula

- $n$ = 100, so: 

$\bar y$  $\sim$ $\frac{\sum_i f_i y_i} {100}$

mean = $\bar y$  $\sim$ $\frac{2887.5} {100}$ = 28.875